sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -125,6 +125,12 @@ private[hive] object HiveShim {
     // for Serialization
     def this() = this(null)
 
+    override def equals(other: Any): Boolean = other match {
+      case a: HiveFunctionWrapper =>
+        functionClassName == a.functionClassName
```

Comment:
@dilipbiswal I feel it is general fine to compare the class name (it is what we do in 1.5. In 1.5, instance is not a part of the constructor). We added instance when we added the support of hive's GenericUDFMacro. I guess we can add a check to see if `GenericUDFMacro` is the class name. If so, we also check instance. Otherwise, we just use the class name. Also, I guess you need to override hashCode as well.